### ISSUE TITLE
Code this plugin

### ISSUE THREAD

**Issue by erocs** [Civcraft/CivDisease#1]

**Comment by sepiatonal** [comment]
How's it supposed to work? I'm interested

**Comment by ghost** [comment]
There was a design thread somewhere on reddit... PRobably need to dig
in changelogs to find it.

Ultimately, I think CivDisease needs some serious rethinking before it
can present a useful addition to Civcraft.

**Comment by ttk2** [comment]
agreed, higher pop is a pre-req, sharding might make it interesting.

**Comment by MonsWTF** [comment]
CivDisease would most likely be very dependent on mechanics that don't constantly harrass the players, but creates a buildup over time that would cause for negative side-effects of not having good 'medical facilities'. I don't think that the plugin is population-specific, but there could certainly be some very interesting mechanics based around human contact via trading, PvP, etc. 

I can imagine this tying into FrankenMod/Contraptions with a Radius-based Gadget that would decrease the 'Progress' of acquiring a disease every hour or so (they could have a few options for radius size, which would mean less factories and more people being covered, but at a marginally higher cost). Such contraptions could be tiered to allow diseases that can't be stopped by a basic facility.

Other than that, I believe there was also the idea that drugs and antibiotics could be produced to combat them, but those could be framed as instant, but more costly, expenses. Vaccines were also discussed, perhaps also high-cost to create, but then low-cost to re-produce them over and over.

Immediate benefits of CivDisease would mean that the "FARMERS" of the server would require some amount of medical attention, and couldn't survive being in crop fields forever.

**Comment by sashimii** [comment]
CivDisease should ideally be renamed so that the subreddit doesn't throw a fit like it did with BergeyPVP. Maybe even repurposed slightly if it can get people more interested in developing it.

Perhaps it can be a mod that lets you easily play around with status effects, and consists of both positive and negative effects. This could potentially allow for simulating all sorts of things: contagious diseases, negative externalities (bastion pollution perhaps), black magic (consider positive effects like speed for a player while other players in near vicinity get slow), and biome-specific effects (like a decrease in saturation the desert). On the other hand, you could also have positive effects (like "health benefits), or perks/passive buffs that are randomly assigned to players upon first login that stay with them and never change. It would make the mod an easier sell to the subreddit.

**Comment by MonsWTF** [comment]
I guess if it were to be re-branded, then perhaps CivMed, or CivMedicine, would be a better name to encompass its focus in the game. There's bad bacteria/viruses/diseases. There's some good bacteria that's rare (very rare probably). There's Factories/Contraptions you can make that mitigate the contraction of all of these, as well as expensive pharmaceuticals, and maybe even long-term vaccinations that remove the possibility of getting a specific disease on your account for IRL months.

I think back when CivDisease was first brought up, there was a very brief discussion on the possibility of buffs to the player, but I do not think those should have a serious mention in this plugin. Any sorts of buffs that directly skew the balance of combat can have the potential to also infuriate the playerbase. I believe it is best to only allow the negatives of deteriorating health from a disease to effect combat, since it is contracted on the long-scale and is not controllable by either participant (Aside for, quite simply, preventing it from happening). 

Development on CivMed should be careful not to make it so that every player wants to carry around a 'Haste' bacteria inside them, or whatever other buffs/debuffing powers they could weild, and the plugin gets flipped on its intended purpose; adding the healthiness aspect of civilization into CivCraft, which is something that is most often ignored due to the carelessness Minecraft allows your character (accidental deaths serving almost no consequence, restocking health only when you need to eat, botting, etc.). The possibility of contracting a harmful disease, and letting players create ways to counteract it, can add a level of game play that does not draw on grinding to solve it, but on figuring out which of those ways is best for your group/city to answer the problem. Build your own healthcare system? Rely on someone else's? Import the necessary medicines/drugs/vaccines to avoid anything serious?

**Comment by sashimii** [comment]
Another thing: simulating [herd immunity](https://en.wikipedia.org/wiki/Herd_immunity). 

Preferably through an elegant solution that doesn't require a top-down approach, so that individual conditions result in herd immunity, rather than a statistic. 

EDIT: We should shift this conversation to the forum's "Balances and Rules" category. We can probably list a bunch of conditions we want to meet with this plugin and talk about it there.

**Comment by ghost** [comment]
Please no Civ prefix in name. If we used that same terrible naming
practice then Citadel would be called Civtadel, CivBiomes, CivPrison,
CivFactory, CivAlert, CivBastion, etc.

**Comment by rourke750** [comment]
Better Disease Civ simulator 3.7474848

**Comment by sashimii** [comment]
PharmaCraft

When you say it out loud people will think it has to do with farming.

**Comment by MonsWTF** [comment]
Or just PharmaC.

And yes I do believe herd immunity is possible without specifically coding herd immunity. Because of the small Civ population, the only ones in direct line of spreading an existing disease would be traders/travelers/fighters. If they get vaccinated, they effectively prevent the rest of their city from contracting whatever diseases they could transfer.

And yeah, this should get moved to Balance and Rules so it can have more awareness from the Subreddit and Discourse.

**Comment by ttk2** [comment]
yes a discourse thread would be nice.
